model/project.go

```diff
@@ -1631,6 +1633,115 @@ func (p *Project) GetDisplayTask(variant, name string) *patch.DisplayTask {
 	return nil
 }
 
+func GetVersionsWithOptions(projectId string, opts GetVersionsOptions) ([]Version, error) {
+	match := bson.M{
+		VersionIdentifierKey: projectId,
+	}
+	if opts.ByBuildVariant != "" {
+		match[bsonutil.GetDottedKeyName(VersionBuildVariantsKey, VersionBuildStatusVariantKey)] = opts.ByBuildVariant
+	}
+	if opts.VersionToStartAt != "" {
+		v, err := VersionFindOneId(opts.VersionToStartAt)
+		if err != nil {
+			return nil, errors.Wrapf(err, "problem finding version '%s'", opts.VersionToStartAt)
+		}
+		if v == nil {
+			return nil, errors.Errorf("version '%s' doesn't exist", opts.VersionToStartAt)
+		}
+		match[VersionCreateTimeKey] = bson.M{"$lte": v.CreateTime}
```

Comment:
if you do lt instead of lte I think you can avoid dropping the pagination version in the route